spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java

```diff
@@ -307,12 +309,17 @@ public LocaleContextResolver localeContextResolver() {
 
 		@Bean
 		@ConditionalOnMissingBean(name = WebHttpHandlerBuilder.WEB_SESSION_MANAGER_BEAN_NAME)
-		public WebSessionManager webSessionManager() {
+		public WebSessionManager webSessionManager(
+				@Autowired(required = false) WebSessionIdResolver webSessionIdResolver) {
```

Comment:
For consistency, it would be good to use `ObjectProvider` rather than `@Autowired(required = false)` here. `getIfAvailable()` can then be used to get the provided ID resolver.

Reply:
Done it.